### DISCUSSION TITLE
Change configuration MaxConcurrencty when appsettings.json changes

### DISCUSSION THREAD

**Discussion by fsaffioti** [quartznet/quartznet#1184] [Q&A]
when I edit my file appsettinsgs.json i got an event when the configuration file is changed
I would like to know how can i change in this case Quartz configuration without restarting the server ?

I was thinking something like this
```
        private void InvokeChanged(IWebHostEnvironment env, IApplicationBuilder app)
        {
            var QuartzOptions = app.ApplicationServices.GetService<IServiceCollectionQuartzConfigurator>();
            var mainAppSettings = Configuration.GetSection(AppConfig.SectionName);
            QuartzOptions.MisfireThreshold = TimeSpan.FromSeconds(mainAppSettings.GetValue<int>("JobMisfireThresholdSec"));
            QuartzOptions.UseDefaultThreadPool(tp =>
            {
                tp.MaxConcurrency = mainAppSettings.GetValue<int>("MaxConcurrency");
            });
        }
```
but app.ApplicationServices.GetService<IServiceCollectionQuartzConfigurator>(); return null
And also, how can I read this configuration?

**Comment by lahma** [comment]
Unfortunately Quartz currently doesn't support such hot reload. When application starts Quartz reads the configuration and initializes its structures like thread pools with the values that were present at initialization time.

**Reply by fsaffioti** [reply]
Thank you for your quick reply, is it possibile to read the settings as MaxConcurrency?

**Reply by lahma** [reply]
If you look at the [appSettings.json](https://github.com/quartznet/quartznet/blob/main/src/Quartz.Examples.AspNetCore/appsettings.json) in example project you can see that you can use any configuration property you want. The properties are documented [here](https://www.quartz-scheduler.net/documentation/quartz-3.x/configuration/reference.html#threadpool).

Check the example project's Startup.cs how configuration is being read.

**Reply by fsaffioti** [reply]
@lahma yeah sorry, I was probably unclear. After I have configured Quartz, I need to query some information from it (for example, MaxConcurrency property). I asked if it is possible to query  MaxConcurrency property because I didn't find any property from SchedulerFactory or Scheduler. Or can I only read MaxConcurrency from my file settings or Configuration? I'm asking this because if the configuration changed:  if it's a Quartz parameter, I'm going to restart the app (since there is not possibility of hot reload); if it's not, I will do something else
Or can I use [Quartz.Shutdown() ](https://quartznet.sourceforge.io/apidoc/3.0/html/html/b9bb5833-2323-88a7-5568-4f8ecf200b76.htm) and then [Quartz.Start()](https://quartznet.sourceforge.io/apidoc/3.0/html/html/aa632b84-a0de-dae9-3e01-a767ebdd29be.htm) for reloading settings?
